### PR TITLE
Initial work on redirect handler

### PR DESCRIPTION
**Pull request by jonsherrard**
#### What have you done
Added server-side redirect paths for Tapestry. Validator for the config and tests.

#### Why have you done it
Projects should be able to configure server-side only redirects.

#### Testing carried out to prevent breaking changes
New tests written, all tests passing. Testing is great. This _demonstrably works_ without unintended side-effects.

### CODE REVIEW

**Comment by jonsherrard** [comment]
Could do with benchmarking Hapi.js. How do request times increase as you register more route handlers. How many redirects could by handled by Tapestry itself?

**Comment by jonsherrard** [comment]
I ran a benchmark on Hapi.js for differences in request speed when you register a lot of routes  and handlers:

https://gist.github.com/333d049a5ddf50ebb874c5436f3198eb

```
Server running at: http://localhost:3006
Server running at: http://localhost:3005
normalRoutes/firstRoute x 302 ops/sec ±5.00% (1000 runs sampled)
highRoutes/firstRoute x 319 ops/sec ±3.04% (1000 runs sampled)
======================================
Fastest is highRoutes/firstRoute

normalRoutes/lastRoute x 300 ops/sec ±3.93% (1000 runs sampled)
highRoutes/lastRoute x 286 ops/sec ±3.61% (1000 runs sampled)
```

Registering 25000 routes, had a negligible effect on selecting the 25000th route, and none on selecting the 1st route. The only slow part is actually registering them on boot, but I think with a config instead of a while loop, you could set them asyncronously.

You'd also only register them in production

```js
import redirects from './config/redirect.js'
const tapestryConfig = {
  redirectPaths: isProd() ? redirects : {}
}
```

**Comment by ecoad** [comment]
Great work!

**Comment by jonsherrard** [comment]
@thomasdigby Any reserverations?

File: test/tests/redirects.test.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { expect } from 'chai'
+import request from 'request'
+import { bootServer, mockProxy, mockApi } from '../utils'
+
+describe('Handling redirects', () => {
+
+  let tapestry = null
+  let config = {
+    redirectPaths: {
+      '/redirect/from/this-path': '/about/home'
```
Comment (thomasdigby) [inline test/tests/redirects.test.js]: Just a question about this API, would an array with `from` and `to` be cleaner? Or unnecessarily verbose?
```
redirectPaths: [{
  from: '/',
  to: '/'
}]
```

Comment (jonsherrard) [inline test/tests/redirects.test.js]: I did think that, but then I also thought.... what if you want 25000 redirects in a file, that'd  add a lot of extra file size.

Using `from` and `to` we could probably validate the config with joi too.

If you have multiple:

```
redirectPaths: [{
  from: '/',
  to: '/'
}, {
 from: '/',
  to: '/'
}, {
 from: '/',
  to: '/'
}, {
 from: '/',
  to: '/'
}, {
 from: '/',
  to: '/'
}, {
 from: '/',
  to: '/'
}]

I think 

{
  '/redirect/from/this-path': '/about/home',
  '/redirect/from/this-path': '/about/home',
  '/redirect/from/this-path': '/about/home',
  '/redirect/from/this-path': '/about/home',
  '/redirect/from/this-path': '/about/home',
  '/redirect/from/this-path': '/about/home'
}
```

Looks a bit neater? 

IDK, something we could bikeshed for a while... for an advanced config feature. I'm easy either way.

Comment (thomasdigby) [inline test/tests/redirects.test.js]: Yep totally agree 👍  happy for you to merge.

**Pull request merged**